main.js

```diff
@@ -662,30 +663,85 @@ function _select_Gateway() { // Check for Gateway used to
-     * prof - profession name
-     * newProfile - new profile definition.
-     * baseProfile - name of base profile to extend. Default is "default" or "XP" in case of Leadership
+     adds new profession and new profile to task list.
+     
+     profession 
+       - name as string of profession to add.  Will create new one if not found
+       - object. will extend or create new profession 
+       Will extend default definition- > can be short
+       
+     profile 
+       - new profile to add.  Set recursiveList to false not use resursive list generation
+       Will extend default definition- > can be short
+       
+    base 
+      - name as string of prifile to use as extension. 
+      Will extend default definition- > can be short
      */
-
-    function addProfileToDefined(prof, newProfile, baseProfile) {
-        baseProfile = baseProfile || (prof === "Leadership" ? "XP" : "default");
-        var profSet = definedTask[prof];
-        if (!profSet) {
-            return;
-        }
-        var bp = {};
-        for (var entry in profSet.profiles) {
-            if (profSet.profiles[entry] && profSet.profiles[entry].profileName == baseProfile) {
-                bp = profSet.profiles[entry];
-                break;
-            }
-        }
-        var cloneBase = jQuery.extend(true, {}, bp);
-        jQuery.extend(true, cloneBase, newProfile);
-        profSet.profiles.push(cloneBase);
+function addProfile(profession, profile, base)
+    {    
+        maxLevel = maxLevel || 25;
+        definedTask = definedTask || {};
+        //general prototype for profession
+        var professionBase = {
+            taskListName: profession, // Friendly name used at the UI
+            taskName: profession, // String used at the gateway
+            taskDefaultPriority: 2, // Priority to allocate free task slots: 0 - High, 1 - Medium, 2 - Low
+            taskActive: true,
+            taskDefaultSlotNum: 0,
+            taskDescription: "",
+            profiles: []  
+        };
+        
+        //profile prototype
+        var profileBase = {
+            profileName : 'Add profile name',
+            isProfileActive : true,
+            recursiveList : true,
+            level : {}
+        };
+        
+      //creating new profession or using existing one 
+    var professionSet =  (typeof profession === 'object') 
+                            ? jQuery.extend(true,professionBase, profession) 
+                            : definedTask[profession] || professionBase;
+      if (!professionSet) { return;}
+        
+      if (!definedTask[profession]){
+        definedTask[profession] = professionSet;
+      }  
+      
+      if (!profile) {return;}
+        
+      //getting base one to extend
+     var newProfile = jQuery.extend(true, {}, profileBase);
+     
+     if (!base) { base = (professionSet.taskListName === 'Leadership' ? 'XP' : 'default');}

```

Comment:
Please avoid using 1-lined if() like that. Something like this instead is easier to read :-)
```javascript
if (!base) 
    base = professionSet.taskListName === 'Leadership' ? 'XP' : 'default';
```